src/libsync/propagatedownload.h

```diff
@@ -15,33 +15,134 @@
 
 #include "owncloudpropagator.h"
 #include "networkjobs.h"
+#include "propagatecommonzsync.h"
 
 #include <QBuffer>
 #include <QFile>
 
 namespace OCC {
 
+class GETJob : public AbstractNetworkJob
+{
+    Q_OBJECT
+protected:
+    QByteArray _etag;
+    time_t _lastModified;
+    QString _errorString;
+    SyncFileItem::Status _errorStatus;
+    bool _bandwidthLimited; // if _bandwidthQuota will be used
+    bool _bandwidthChoked; // if download is paused (won't read on readyRead())
+    qint64 _bandwidthQuota;
+    QPointer<BandwidthManager> _bandwidthManager;
+
+public:
+    GETJob(AccountPtr account, const QString &path, QObject *parent = 0) :
+        AbstractNetworkJob(account, path, parent)
+        , _etag(NULL)
+        , _lastModified()
+        , _errorStatus(SyncFileItem::NoStatus)
+        , _bandwidthLimited(false)
+        , _bandwidthChoked(false)
+        , _bandwidthQuota(0)
+        , _bandwidthManager(0)
+    {
+    }
+
+    ~GETJob()
+    {
+        if (_bandwidthManager) {
+            _bandwidthManager->unregisterDownloadJob(this);
+        }
+    }
+
+    virtual qint64 currentDownloadPosition() = 0;
+    virtual quint64 resumeStart() { return 0; }
+
+    QByteArray &etag() { return _etag; }
+    time_t lastModified() { return _lastModified; }
+
+    void setErrorString(const QString &s) { _errorString = s; }
+    QString errorString() const;
+    SyncFileItem::Status errorStatus() { return _errorStatus; }
+    void setErrorStatus(const SyncFileItem::Status &s) { _errorStatus = s; }
+    void setBandwidthManager(BandwidthManager *bwm);
+    void setChoked(bool c);
+    void setBandwidthLimited(bool b);
+    void giveBandwidthQuota(qint64 q);
+    void onTimedOut();
+
+signals:
+    void finishedSignal();
+};
+
 /**
- * @brief The GETFileJob class
+ * @brief Downloads the zsync metadata and uses the original file as a seed, then downloads needed ranges via GET
  * @ingroup libsync
  */
-class GETFileJob : public AbstractNetworkJob
+class GETFileZsyncJob : public GETJob
+{
+    Q_OBJECT
```

Comment:
add `private:` here? I had to look up the default visibility (and here @ogoffart laughing), so making it explicit might save devs time in the future. Thanks... and yes, it's minor ;-)